src/git_client/git_cli.rs

```diff
@@ -18,12 +20,55 @@ impl GitCli {
     }
 }
 
+/// Implementation of the `GitClient` trait for the `GitCli` struct.
 impl GitClient for GitCli {
+    /// Performs an interactive commit.
     fn interactive_commit(&self) {
         println!("Committing changes");
     }
 
+    /// Checks out the specified branch.
+    ///
+    /// # Arguments
+    ///
+    /// * `branch` - The name of the branch to check out.
     fn checkout(&self, branch: &str) {
         println!("Checking out branch: {}", branch);
     }
+
+    /// Retrieves information about the current Git repository.
+    ///
+    /// # Returns
+    ///
+    /// A `Result` containing a tuple with three elements: the provider, owner, and repository name.
+    ///
+    /// # Errors
+    ///
+    /// This function may return an error if:
+    ///
+    /// * The `git` command fails to execute.
+    /// * The current directory is not a Git repository.
+    fn get_repo_info(&self) -> Result<(String, String, String)> {
+        let out = process::Command::new("git")
+            .arg("remote")
+            .arg("-v")
+            .output()
+            .context("Failed to get git remote")
+            .suggestion("Make sure you are inside a git repository")?
+            .stdout
+            .iter()
+            .map(|&c| c as char)
+            .collect::<String>();
+
+        let mut parts = out
+            .split(['/', '@', ':', '.'])
+            .collect::<Vec<&str>>();
+        parts.retain(|&s| !s.is_empty());
```

Comment:
I've tested it and it works for both, that's why I'm using all those separators. Also, the retain method is mostly there for the http format, because the doubled '/' at the beginning would cause the vector to contain empty strings.

Though, I can make it more obvious which format I'm working with.

Reply:
Ok, then you don't need to change it. My bad.